corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -25,7 +24,13 @@ def JSON(obj):
     # json.dumps does not properly convert QueryDict array parameter to json
     if isinstance(obj, QueryDict):
         obj = dict(obj)
-    return mark_safe(json.dumps(obj, default=json_handler))
+    try:
+        return mark_safe(json.dumps(obj, default=json_handler))
+    except TypeError as e:
+        msg = ("Unserializable data was sent to the `|JSON` template tag.  "
+               "If DEBUG is off, Django will silently swallow this error.")
```

Comment:
`notify exception` includes [a stack trace](https://github.com/dimagi/dimagi-utils/blob/ab7e792c47c3aeef60072b47cf5bc32ed6d472be/dimagi/utils/logging.py#L37).  Backoff sounds nice though.  I'm hesitant to include `obj` in the email because it includes client data, and I figure the stack trace will be enough to track down the issue.  There error message does tell you something about the object, however:
```
TypeError: <SQLLocation(domain=ethan-commtrack, name=Atlantic Beach)> is not JSON serializable
```